src/menu.tsx

```diff
@@ -61,115 +230,181 @@ const Menu: React.FC<MenuProps> = ({
+            }
+
+            const nextFocusedItem = reverse && nextItem !== -1 ? items.length - 1 - nextItem : nextItem;
+            setFocusedItem(nextFocusedItem < 0 ? null : nextFocusedItem);
+            items[nextItem]?.focus();
+        },
+        [focusedItem, menu]
     );
 
+    React.useEffect(() => {
+        if (!isMenuOpen) {
+            setFocusedItem(null);
+        } else if (isOpenedwithKeyboard && menu) {
+            setFirstFocusableItem();
+            setIsOpenedwithKeyboard(false);
+        }
+    }, [isMenuOpen, setFirstFocusableItem, isOpenedwithKeyboard, menu]);
+
     React.useEffect(() => {
         const handleKeyDown = (e: KeyboardEvent) => {
             if (isMenuOpen) {
-                if (e.keyCode === TAB) {
-                    cancelEvent(e);
+                switch (e.keyCode) {
+                    case RIGHT:
+                    case DOWN:
+                        setNextFocusableItem();
+                        e.preventDefault();
+                        e.stopPropagation();
+                        break;
+                    case LEFT:
+                    case UP:
+                        setNextFocusableItem(true);
+                        e.preventDefault();
+                        e.stopPropagation();
+                        break;
+                    case ESC:
+                        setIsMenuOpen(false);
+                        break;
+                    case SPACE:
+                    case ENTER:
+                        cancelEvent(e);
+                        if (focusedItem !== null) {
+                            getMenuItems(menu)[focusedItem].click();
+                        }
+                        break;
+                    case TAB:
+                        cancelEvent(e);
+                        break;
+
+                    default:
+                    // do nothing
                 }
-                if (e.keyCode === ESC) {
-                    setIsMenuOpen(false);
+            } else {
+                switch (e.keyCode) {
+                    case ENTER:
+                    case SPACE:
+                        setIsOpenedwithKeyboard(true);
+                        break;
+                    case DOWN:
+                        if (target === document.activeElement) {
+                            setIsOpenedwithKeyboard(true);
+                            setIsMenuOpen(true);
+                        }
+                        break;
+                    default:
+                    // do nothing
                 }
             }
         };
+
         document.addEventListener('keydown', handleKeyDown, false);
         return () => {
             document.removeEventListener('keydown', handleKeyDown, false);
         };
     });
 
+    React.useEffect(() => {
+        if (menu) {
+            menu.setAttribute('role', 'menu');
+            menu.setAttribute('aria-expanded', String(isMenuOpen));
```

Comment:
`aria-expanded` is being currently added to the menu container, but it should be in the trigger (touchable / button)

Reply:
good catch. Fixed